src/figuresClasses.ts

```diff
@@ -1,19 +1,112 @@
-export interface Figure {
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export abstract class BaseFigure implements Figure {
+  abstract readonly shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    const arrayOfSides = [a, b, c];
+    const isSomeSideNegative:boolean = arrayOfSides.some(
```

Comment:
No need to specify type here, TS understand that there will be boolean value since it's primitive

Reply:
And you've written similar logic in all classes, would be logical to create helper method for it and reuse it